libraries/typescript/Microsoft.Bot.Builder.Adapters.WeChat/src/weChatAdapter.ts

```diff
@@ -0,0 +1,373 @@
+     * Key to get all response from bot in a single turn.
+     */
+    private TurnResponseKey = 'turnResponse';
+
+    private weChatMessageMapper: WeChatMessageMapper;
+    private weChatClient: WeChatClient;
+    protected readonly settings: WeChatAdapterSettings;
+
+    /**
+     * Creates an instance of we chat adapter.
+     * @param storage
+     * @param settings configuration settings for the adapter.
+     */
+    constructor(storage: Storage, settings: WeChatAdapterSettings) {
+        super();
+        this.settings = {
+            AppId: '',
+            AppSecret: '',
+            Token: '',
+            EncodingAESKey: '',
+            UploadTemporaryMedia: undefined,
+            ...settings
+        };
+        this.weChatClient = new WeChatClient(this.settings.AppId, this.settings.AppSecret, storage);
+        this.weChatMessageMapper = new WeChatMessageMapper(this.weChatClient, this.settings.UploadTemporaryMedia);
+    }
+
+    /**
+     * Resume a conversation with a user, possibly after some time has gone by.
+     * @param reference A `ConversationReference` saved during a previous incoming activity.
+     * @param logic A function handler that will be called to perform the bots logic after the the adapters middleware has been run.
+     * @returns conversation
+     */
+    public async continueConversation(reference: Partial<ConversationReference>, logic: (context: TurnContext) => Promise<void>): Promise<void> {
+        const request: Partial<Activity> = TurnContext.applyConversationReference({ type: 'event', name: 'continueConversation' }, reference,true);
+        const context: TurnContext = this.createContext(request);
+
+        await this.runMiddleware(context, logic as any);
+    }
+
+    /**
+     * Allows for the overriding of the context object in unit tests and derived adapters.
+     * @param request Received request.
+     */
+    protected createContext(request: Partial<Activity>): TurnContext {
+        return new TurnContext(this as any, request);
+    }
+
+    /**
+     * Process the request from WeChat.
+     * @param wechatRequest Request message entity from wechat.
+     * @param logic A function handler that will be called to perform the bots logic after the the adapters middleware has been run.
+     * @param passiveResponse Marked the message whether it needs passive reply or not.
+     * @returns Response message entity.
+     */
+    public async ProcessWeChatRequest(wechatRequest: IRequestMessageBase, logic: (context: TurnContext) => Promise<any>, passiveResponse: boolean): Promise<any> {
+        const activity = await this.weChatMessageMapper.ToConnectorMessage(wechatRequest);
+        const context = new TurnContext(this, activity as Activity);
+        const responses = new Map<string, Array<Activity>>();
+        context.turnState.set(this.TurnResponseKey, responses);
+        await this.runMiddleware(context, logic);
+        const key = `${ activity.conversation.id }:${ activity.id }`;
```

Comment:
>const key = `${ activity.conversation.id }:${ activity.id }`; [](start = 8, length = 61)

do you really need to do this?  turnState is unique per TurnContext so each processWeChatRequest() call will have its own turnState. That means the responses will ONLY ever be for the current activity. Seems like you could simply store an [] under turnResponseKey and then just append all responses to that array. I've done this before myself and it works fine.